extensions/interactions/ItemSelectionInput/validator.js

```diff
@@ -89,15 +89,60 @@ oppia.filter('oppiaInteractiveItemSelectionInputValidator', [
         baseInteractionValidationService.getAnswerGroupWarnings(
           answerGroups, stateName));
 
-      if (!defaultOutcome ||
-          $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+      var selectedChoices = [];
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              var choicePreviouslySelected = (
```

Comment:
Note that, for ItemSelectionInput, there's a rule called "DoesNotContainAtLeastOneOf" for which you would regard the non-specified choices as selected, so the logic of taking whatever is specified in the rule and marking off those choices does not work in all cases.